MSMAll/scripts/MSMAll.sh

```diff
@@ -1239,37 +1213,17 @@ main()
 		case ${MatlabRunMode} in
 			0)
 				# Use Compiled Matlab
-				matlab_exe="${HCPPIPEDIR}"
-				matlab_exe+="/MSMAll/scripts/Compiled_MSMregression/run_MSMregression.sh"
+				matlab_exe="${HCPPIPEDIR}/MSMAll/scripts/Compiled_MSMregression/run_MSMregression.sh"
 
 				matlab_compiler_runtime="${MATLAB_COMPILER_RUNTIME}"
 
-				matlab_function_arguments="'${inputspatialmaps}'"
-				matlab_function_arguments+=" '${inputdtseries}'"
-				matlab_function_arguments+=" '${inputweights}'"
-				matlab_function_arguments+=" '${outputspatialmaps}'"
-				matlab_function_arguments+=" '${outputweights}'"
-				matlab_function_arguments+=" '${Caret7_Command}'"
-				matlab_function_arguments+=" '${Method}'"
-				matlab_function_arguments+=" '${Params}'"
-				matlab_function_arguments+=" '${VN}'"
-				matlab_function_arguments+=" ${nTPsForSpectra}"
-				matlab_function_arguments+=" '${BC}'"
-   				matlab_function_arguments+=" '${VolParams}'"
-
-				matlab_logging=">> ${StudyFolder}/${Subject}.MSMregression.matlab.1.log 2>&1"
-
-				matlab_cmd="${matlab_exe} ${matlab_compiler_runtime} ${matlab_function_arguments} ${matlab_logging}"
-
-				# Note: Simply using ${matlab_cmd} here instead of echo "${matlab_cmd}" | bash
-				#       does NOT work. The output redirects that are part of the ${matlab_logging}
-				#       value, get quoted by the run_*.sh script generated by the MATLAB compiler
-				#       such that they get passed as parameters to the underlying executable.
-				#       So ">>" gets passed as a parameter to the executable as does the
-				#       log file name and the "2>&1" redirection. This causes the executable
-				#       to die with a "too many parameters" error message.
-				log_Msg "Run Matlab command: ${matlab_cmd}"
-				echo "${matlab_cmd}" | bash
+				matlab_function_arguments=("${inputspatialmaps}" "${inputdtseries}" "${inputweights}" "${outputspatialmaps}" "${outputweights}" "${Caret7_Command}" "${Method}" "${Params}" "${VN}" "${nTPsForSpectra}" "${BC}" "${VolParams}")
+
+				matlab_cmd=("${matlab_exe}" "${matlab_compiler_runtime}" "${matlab_function_arguments[@]}")
+
+				#don't log to a separate file, separate log files have never been desirable
+				log_Msg "Run Matlab command: ${matlab_cmd[*]}"
+				echo "${matlab_cmd[@]}" | bash
```

Comment:
Why are we invoking compiled matlab different here (by piping to bash) compared to the other instances in the script?

Reply:
Good catch, apparently I lost track of the edits to all the separate matlab sections.